### PR TITLE
[refactor]: Core, `sumeragi`, instance functions, `torii`

### LINKED ISSUE #1812: Add warnings when `iroha` is given unrecognised command-line flags.
Currently, the Iroha peer binary ignores the command line flags that are incorrect. Since the number of flags is limited, it should be possible to provide an efficient sanity check for the end user, warning of any unrecognised flags.

### PR DESCRIPTION
**Pull request by appetrosyan**
Signed-off-by: Aleksandr <a-p-petrosyan@yandex.ru>

### Description of the Change

Aggregate of several refactors to `Sumeragi`, `Torii` and various fixes to `Core`. 
- Lints introduced in rust 1.59
- Sumeragi split into file-based modules. 
- Torii and related functionality split off from the `iroha_core` crate. 
- Iroha peer binary also moved into the front-end crate. 
- Various CLI fixes and error messages. 

### Issue

Closes #1812 
Opens #1975 

### Benefits

Better organisation

### Possible Drawbacks

Large pull request

### CODE REVIEW

File: lints.toml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#
+    #
 # For all clippy lints please visit: https://rust-lang.github.io/rust-clippy/master/
 #
```
Comment (s8sato) [inline lints.toml:3]:
```suggestion
# For all clippy lints please visit: https://rust-lang.github.io/rust-clippy/master/

```

File: cli/src/torii/routing.rs [hunk 2]
```diff
@@ -32,9 +36,11 @@ pub(crate) async fn handle_queries<W: WorldTrait>(
...
     pagination: Pagination,
     request: VerifiedQueryRequest,
-) -> Result<Scale<VersionedQueryResult>, query::Error> {
-    let valid_request = request.validate(&wsv, &query_validator)?;
-    let result = valid_request.execute(&wsv)?;
+) -> Result<Scale<VersionedQueryResult>, warp::http::Response<warp::hyper::Body>> {
+    let valid_request = request
+        .validate(&wsv, &query_validator)
+        .map_err(into_reply)?;
+    let result = valid_request.execute(&wsv).map_err(into_reply)?;
```
Comment (s8sato) [inline cli/src/torii/routing.rs:43]: No need to `map_err` as `query::Error` will be recovered into `Response` in `handle_rejection`.

However, if we have every handler have `Response<Body>` Err variant,
`handle_rejection` will be one-line -- just pick `Response<Body>`.
We could go this way

Comment (appetrosyan) [reply cli/src/torii/routing.rs:43]: You mean impl From? I would do that, but it introduced a bidirectional dependency. Since this was the only place where `impl Reply` was used, I decided to inline the implementation.

Comment (appetrosyan) [reply cli/src/torii/routing.rs:43]: Response<Body> actually looks interesting.

File: core/src/sumeragi/config.rs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        public_key: PublicKey::default(),
+    }
+}
+
+// `HashSet::new()` is not const yet.
+fn default_empty_trusted_peers() -> TrustedPeers {
+    TrustedPeers {
+        peers: HashSet::new(),
+    }
+}
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply core/src/sumeragi/config.rs:172]: Since the default for a primitive type is set, (and often zero), if we want the default to be anything but 0, we need a new type, with an `impl Default` for each. I'm not sure it's more compact that way.

File: cli/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+use std::collections::BTreeMap;
+
+use iroha_data_model::prelude::*;
+
+/// Returns the `domain_name: domain` mapping, for initial domains.
+///
+/// # Errors
+/// - Genesis account public key not specified.
+pub fn domains(configuration: &config::Configuration) -> Result<BTreeMap<DomainId, Domain>> {
```
Comment (mversic) [inline cli/src/lib.rs:308]:
```suggestion
fn domains(configuration: &config::Configuration) -> Result<BTreeMap<DomainId, Domain>> {
```

File: cli/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+        )>,
+        _config: &Configuration,
+    ) -> Result<bool> {
+        Ok(false)
+    }
+}
+
+use std::collections::BTreeMap;
+
+use iroha_data_model::prelude::*;
```
Comment (mversic) [inline cli/src/lib.rs:302]: move use statements to the top of the module

File: cli/src/torii/tests.rs [hunk 6]
```diff
@@ -547,6 +569,28 @@ async fn query_with_no_find() {
         .assert()
 }
 
+// THIS SHOULD NOT BE POSSIBLE!!!
+pub fn new_dummy() -> ValidBlock {
```
Comment (mversic) [inline cli/src/torii/tests.rs:573]:
```suggestion
fn new_dummy() -> ValidBlock {
```

File: client/Cargo.toml [hunk 7]
```diff
@@ -39,6 +39,9 @@ base64 = "0.13.0"
 [dev-dependencies]
 iroha_core = { version = "=2.0.0-pre-rc.2", path = "../core", features = ["dev-telemetry", "telemetry"] }
 iroha_permissions_validators = { version = "=2.0.0-pre-rc.2", path = "../permissions_validators" }
+iroha = { path = "../cli", features = ["dev-telemetry", "telemetry"] }
+iroha_data_model = { version = "=2.0.0-pre-rc.2", path = "../data_model", features = ["warp", "cross_crate_testing"] }
```
Comment (mversic) [inline client/Cargo.toml:43]: the same as `cli/Cargo.toml` comment. Features are probably lumped together

File: client/src/lib.rs [hunk 8]
```diff
@@ -15,12 +15,14 @@ pub mod samples {
...
     /// Get sample client configuration.
+    #[allow(clippy::expect_used)]
     pub fn get_client_config(key_pair: &KeyPair) -> Configuration {
         let (public_key, private_key) = key_pair.clone().into();
         Configuration {
             public_key,
             private_key,
-            account_id: iroha_data_model::prelude::AccountId::test("alice", "wonderland"),
+            account_id: iroha_data_model::prelude::AccountId::new("alice", "wonderland")
+                .expect("Should not fail."),
```
Comment (mversic) [inline client/src/lib.rs:25]: why this change? you consider it more safe?

Comment (mversic) [reply client/src/lib.rs:25]: also, this method is only used in tests and we should find a way to exclude it from release builds

File: core/docs/Cargo.toml [hunk 9]
```diff
@@ -9,6 +9,7 @@ edition = "2021"
 [dependencies]
```
Comment (mversic) [inline core/docs/Cargo.toml:9]: do we make use of the `core/docs`?

File: telemetry/derive/Cargo.toml [hunk 10]
```diff
@@ -26,6 +26,6 @@ proc-macro2 = "1.0"
 proc-macro-error = "1.0"
 
 [dev-dependencies]
-iroha_core = { path = "../../core" }
+iroha_core = { version = "=2.0.0-pre-rc.2", path = "../../core" }
```
Comment (mversic) [inline telemetry/derive/Cargo.toml:29]: managing versions manually like this seems problematic. Could we consider some automated way to bump versions across all crates? also it doesn't seem too important to specify versions for `dev-dependencies`?

File: lints.toml [hunk 11]
```diff
@@ -51,6 +51,7 @@ allow = [
     'elided_lifetimes_in_paths',
     # Our preferred style.
     'clippy::mod-module-files',
+    'clippy::separated-literal-suffix',
```
Comment (mversic) [inline lints.toml:54]: I would actually like it more if the suffix was separated with underscore

File: data_model/src/lib.rs [hunk 12]
```diff
@@ -94,13 +94,16 @@ impl Name {
...
+        Self::from_str(name)
     }
 
     /// Instantly construct [`Name`] assuming `name` is valid.
     #[inline]
     #[allow(clippy::expect_used)]
+    #[cfg(any(test, feature = "cross_crate_testing"))]
     pub fn test(name: &str) -> Self {
+        // I'm not convinced we need these even for testing. Ideally
+        // we should *just* make `Name::new` `const`.
```
Comment (mversic) [inline data_model/src/lib.rs:106]:
```suggestion
        // NOTE: I'm not convinced we need these even for testing. Ideally
        // we should *just* make `Name::new` `const`.
```

Comment (mversic) [reply data_model/src/lib.rs:106]: I agree that we don't need this. I thought it was for creating names without the check akin to `new_unchecked`

Comment (mversic) [reply data_model/src/lib.rs:106]: I'd like if you would remove it and all other `test` methods(like `Domain::test` or `Account::test`) entirely if @s8sato agrees

File: core/src/tx.rs [hunk 13]
```diff
@@ -371,61 +371,50 @@ impl From<RejectedTransaction> for AcceptedTransaction {
...
-            TransactionValidator::new(tx_limits, AllowAll::new(), AllowAll::new(), wsv)
-                .validate(accepted_tx, true)
-                .expect("Failed to validate.")
-                .hash();
-        assert_eq!(tx_hash, signed_tx_hash);
-        assert_eq!(tx_hash, accepted_tx_hash);
-        assert_eq!(tx_hash, valid_tx_hash.transmute());
-    }
+    // #[test]
+    // fn hash_should_be_the_same() {
```
[the remark this answers is not in the document]
Comment (mversic) [reply core/src/tx.rs:380]: you could use `#[ignore]` instead

**Review by mversic — CHANGES_REQUESTED** [review]

File: core/src/sumeragi/config.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (s8sato) [reply core/src/sumeragi/config.rs:172]: Every field is default so deriving should be enough though... am I missing something?

File: cli/Cargo.toml [hunk 14]
```diff
@@ -25,9 +25,29 @@ is-it-maintained-open-issues = { repository = "https://github.com/hyperledger/ir
...
+serde_json = "1.0"
+tokio = { version = "1.6.0", features = ["sync", "time", "rt", "io-util", "rt-multi-thread", "macros", "fs"]}
+warp = "0.3"
+thiserror = "1.0.28"
 color-eyre = "0.5.11"
-tokio = { version = "1.6.0", features = ["rt", "rt-multi-thread"]}
+
+[dev-dependencies]
+unique_port = "0.1.0"
+iroha_core = { version = "=2.0.0-pre-rc.2", path = "../core", features = ["cross_crate_testing"] }
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply cli/Cargo.toml:52]: That's a bug in the dependency resolution, and AFAIK it will be fixed eventually. For now, I think that marking everything that isn't meant for release builds as optional, even if it gets compiled is still an improvement.

[the remark this answers is not in the document]
Comment (mversic) [reply cli/Cargo.toml:52]: One way to mitigate this is to extract code required in tests which needs to be shared among more crates into a separate crate

File: client/Cargo.toml (hunk 7 above)
Comment (appetrosyan) [reply client/Cargo.toml:43]: Actually, let me quickly test it.

File: cli/Cargo.toml (hunk 14 above)
[the remark this answers is not in the document]
Comment (mversic) [reply cli/Cargo.toml:52]: also: https://doc.rust-lang.org/cargo/reference/resolver.html#feature-resolver-version-2

File: client/src/lib.rs (hunk 8 above)
Comment (appetrosyan) [reply client/src/lib.rs:25]: Here's my opinion. Ideally, we shouldn't ever be skipping any checks in either tests or benchmarks. They either have negligible overhead, and having checked and unchecked variants is unnecessary complexity, or they have significant overhead, which we would see in the real world too.

File: core/docs/Cargo.toml (hunk 9 above)
Comment (appetrosyan) [reply core/docs/Cargo.toml:9]: Yes. We check if we added config options but forgot to check them in. Since there's other kinds of documentation, this could be removed, but I want the entire team to sign off on that decision.

File: telemetry/derive/Cargo.toml (hunk 10 above)
Comment (appetrosyan) [reply telemetry/derive/Cargo.toml:29]: I didn't do it manually. The crates are modified in lockstep, and changing `=vX.X.X` to `=vX.X.X+1` is a simple text replacement.

File: core/src/tx.rs (hunk 13 above)
[the remark this answers is not in the document]
Comment (appetrosyan) [reply core/src/tx.rs:380]: Neither. This test doesn't compile, and I was supposed to fix it during the rebase.

File: core/src/sumeragi/config.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (appetrosyan) [reply core/src/sumeragi/config.rs:172]: Every field is default, but default is determined by type. So every field defaults to 0, which is not what we want.

[the remark this answers is not in the document]
Comment (s8sato) [reply core/src/sumeragi/config.rs:172]: I don't get it.
None of `String`, `PublicKey`, `HashSet` has anything to do with the digit 0.
Also any other manual `impl Default` for these structs does not appear to me

File: data_model/src/lib.rs (hunk 12 above)
Comment (s8sato) [reply data_model/src/lib.rs:106]: I don't mind.
What I intended to provide was short writing without `.expect()` or `.unwrap()`

File: core/docs/Cargo.toml (hunk 9 above)
Comment (mversic) [reply core/docs/Cargo.toml:9]: open an issue to discuss?

File: cli/Cargo.toml (hunk 14 above)
[the remark this answers is not in the document]
Comment (appetrosyan) [reply cli/Cargo.toml:52]: Resolver = 2 seems to be what we need.

File: core/docs/Cargo.toml (hunk 9 above)
Comment (appetrosyan) [reply core/docs/Cargo.toml:9]: Yep. And create a poll

Comment (appetrosyan) [reply core/docs/Cargo.toml:9]: Created #1968

File: core/src/sumeragi/config.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (appetrosyan) [reply core/src/sumeragi/config.rs:172]: I think we may have misunderstood each other. Do you want me to remove `impl Default for SumeragiConfiguration` where these functions are used, or just to inline `default_peer_id`?

File: cli/src/lib.rs [hunk 15]
```diff
@@ -0,0 +1,320 @@
...
+    /// Queue of transactions
+    pub queue: Arc<Queue<W>>,
+    /// Sumeragi consensus
+    pub sumeragi: AlwaysAddr<S>,
+    /// Kura - block storage
+    pub kura: AlwaysAddr<K>,
+    /// Block synchronization actor
+    pub block_sync: AlwaysAddr<B>,
+    /// Torii web server
+    pub torii: Option<Torii<W>>,
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply cli/src/lib.rs:81]: [On line 244](https://github.com/hyperledger/iroha/pull/1965/files/1b731973880e5f0c229e1f0d3202bab3ab170376#diff-c1f8f7498da827a634bddc8a7559198bc99b296e9d9e8b91a70b503662995b8cR244) it's used to see if the peer had already been started or not. 

It's a little unorthodox, but gets the job of not letting you have two instances of `torii` done. Any alternative implementation of the same would be less effective.

File: cli/src/lib.rs [hunk 16]
```diff
@@ -0,0 +1,320 @@
...
+            .start()
+            .await
+            .wrap_err("Failed to start Torii")
+    }
+
+    /// Starts iroha in separate tokio task.
+    ///
+    /// # Errors
+    /// - Forwards initialisation error.
+    pub fn start_as_task(&mut self) -> Result<JoinHandle<eyre::Result<()>>> {
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply cli/src/lib.rs:254]: It's used in the test network, I've added the necessary feature flag.

File: cli/src/lib.rs (hunk 15 above)
[the remark this answers is not in the document]
Comment (mversic) [reply cli/src/lib.rs:81]: I see, would you add a small comment in the code please?

File: core/src/sumeragi/config.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (s8sato) [reply core/src/sumeragi/config.rs:172]: Okay, my suggestion is as follows:

`iroha_data_model::peer`
```rust
#[derive(Default)]
pub struct Id {
    //
}
```
`iroha_core::sumeragi::config`
```rust
impl Default for SumeragiConfiguration {
            //
            trusted_peers: TrustedPeers::default(),
            peer_id: PeerId::default(),
            //
}

[the remark this answers is not in the document]
Comment (appetrosyan) [reply core/src/sumeragi/config.rs:172]: Oh! Ok. That makes a lot of sense.

**Review by Arjentix — CHANGES_REQUESTED** [review]
I'll continue my review later. For now checked only 25 files

File: data_model/src/lib.rs (hunk 12 above)
Comment (s8sato) [reply data_model/src/lib.rs:106]: BTW `Name::test` is not unchecked. It panics by wrong input.
I still agree with the removal though

File: data_model/src/expression.rs [hunk 17]
```diff
@@ -810,19 +810,20 @@ pub struct WhereBuilder {
 
 impl WhereBuilder {
     /// Sets the `expression` to be evaluated.
-    pub fn evaluate<E: Into<EvaluatesTo<Value>>>(expression: E) -> WhereBuilder {
-        WhereBuilder {
+    pub fn evaluate<E: Into<EvaluatesTo<Value>>>(expression: E) -> Self {
```
Comment (Arjentix) [inline data_model/src/expression.rs:813]: `must_use`?

File: data_model/src/transaction.rs [hunk 18]
```diff
@@ -276,12 +275,14 @@ impl Transaction {
...
 
     /// Adds metadata to the `Transaction`
+    #[must_use]
     pub fn with_metadata(mut self, metadata: UnlimitedMetadata) -> Self {
         self.payload.metadata = metadata;
         self
     }
 
     /// Adds nonce to the `Transaction`
+    #[must_use]
```
Comment (Arjentix) [inline data_model/src/transaction.rs:285]: What about `must_use` for `new()` and/or `sign()` methods? Can't leave comment for its lines

File: lints.toml [hunk 19]
```diff
@@ -1,4 +1,4 @@
-#
+    #
```
Comment (Arjentix) [inline lints.toml:1]:
```suggestion
#
```

**Review by Arjentix — CHANGES_REQUESTED** [review]

File: client/src/config.rs [hunk 20]
```diff
@@ -99,10 +98,11 @@ pub struct Configuration {
 
 impl Default for Configuration {
     fn default() -> Self {
+        #[allow(clippy::expect_used)]
         Self {
             public_key: PublicKey::default(),
             private_key: PrivateKey::default(),
-            account_id: AccountId::test("", ""),
+            account_id: AccountId::new("", "").expect("Empty strings are valid"),
```
[the remark this answers is not in the document]
Comment (Arjentix) [reply client/src/config.rs:105]: After a full review I see that this is the only place using that and it matches well with other code, so no need to create `default()` or `empty()`

File: cli/src/main.rs [hunk 21]
```diff
@@ -1,24 +1,30 @@
...
         return Ok(());
     }
 
-    if std::env::args().any(|arg| arg == "--submit" || arg == "--submit-genesis" || arg == "-s") {
+    if std::env::args().any(|a| is_submit(&a)) {
         args.submit_genesis = true;
     }
 
+    for arg in std::env::args().skip(1) {
+        if !is_help(&arg) && !is_submit(&arg) {
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply cli/src/main.rs:22]: > Cause it can be painful in future if we will have a lot of complex flags.

We won't. If we do, we'll revert back to using `Clap`. The whole point is not to import a large crate for two flags.

File: core/src/block.rs [hunk 22]
```diff
@@ -536,6 +537,7 @@ impl ValidBlock {
     }
 
     /// Validate block transactions against current state of the world.
+    #[must_use]
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply core/src/block.rs:540]: Not necessarily. Most cases that are caught by the linter are, though.

File: core/src/wsv.rs [hunk 23]
```diff
@@ -294,10 +294,11 @@ impl<W: WorldTrait> WorldStateView<W> {
...
 
-    // TODO: There could be just this one method `blocks` instead of `blocks_from_height` and
-    // `blocks_after_height`. Also, this method would return references instead of cloning
-    // blockchain but comes with the risk of deadlock if consumer of the iterator stores
-    // references to blocks
+    // TODO: There could be just this one method `blocks` instead of
+    // `blocks_from_height` and `blocks_after_height`. Also, this
+    // method would return references instead of cloning blockchain
+    // but comes with the risk of deadlock if consumer of the iterator
+    // stores references to blocks
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply core/src/wsv.rs:301]: @Arjentix Good idea. Create an issue for it. Put into Chores on ZenHub

File: cli/src/lib.rs [hunk 24]
```diff
@@ -0,0 +1,320 @@
...
+    pub config_path: PathBuf,
+}
+
+const CONFIGURATION_PATH: &str = "config.json";
+const GENESIS_PATH: &str = "genesis.json";
+
+impl Default for Arguments {
+    fn default() -> Self {
+        Self {
+            submit_genesis: false,
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply cli/src/lib.rs:49]: It's an unnecessary level of indirection, but I don't see why not.

File: core/src/tx.rs (hunk 13 above)
[the remark this answers is not in the document]
Comment (appetrosyan) [reply core/src/tx.rs]: Fixed now by moving to `torii`.

File: core/src/wsv.rs (hunk 23 above)
[the remark this answers is not in the document]
Comment (Arjentix) [reply core/src/wsv.rs:315]: Done #1974

File: docs/source/references/config.md [hunk 25]
```diff
@@ -595,72 +588,6 @@ Has type `u64`. Can be configured via environment variable `SUMERAGI_TX_RECEIPT_
 500
 ```
 
-## `telemetry`
```
Comment (s8sato) [inline docs/source/references/config.md:598]: I think documentations should cover all the features

Comment (appetrosyan) [reply docs/source/references/config.md:598]: Iroha's feature flags need more attention than they received.

File: core/docs/Cargo.toml [hunk 26]
```diff
@@ -6,9 +6,19 @@ edition = "2021"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+# Explanation: You want the option to remove configuration docs if you distribute e.g. a `deb`, but want the person who just cloned the repo to be able to see at a glance what kinds of features are accessible to them, what they do and how to use them. #1975 should be fixed soon.
```
Comment (s8sato) [inline core/docs/Cargo.toml:9]: If we had some [Build Scripts](https://doc.rust-lang.org/cargo/reference/build-scripts.html) to generate the doc with the build, the features would be synchronized between the docs and the distribution

**Review by s8sato — APPROVED** [review]

File: core/docs/Cargo.toml (hunk 26 above)
Comment (appetrosyan) [reply core/docs/Cargo.toml:9]: Could you add that to the issue?

File: client/src/lib.rs (hunk 8 above)
Comment (mversic) [reply client/src/lib.rs:25]: yeah, I think my comment was offtopic. We can close this. However, it should be fine to use `unwrap` in tests. Also once `Option::expect` or/and `Option::unwrap` are const functions I think this should remove the necessity to use `static Lazy` which I've seen around.

File: client/Cargo.toml [hunk 27]
```diff
@@ -32,13 +32,16 @@ serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 tungstenite = { version = "0.16" }
 smallvec = { version = "1.8.0", features = ["serde", "union"] }
-smallstr = { version = "0.2.0", features = ["serde", "union"] }
+smallstr = { git = "https://github.com/murarth/smallstr", rev = "68efc08", default-features = false, features = ["serde", "union"] }
```
Comment (mversic) [inline client/Cargo.toml:35]: again, use `0.3.0` version`

File: cli/src/torii/routing.rs [hunk 28]
```diff
@@ -319,8 +333,10 @@ pub(crate) async fn handle_rejection(rejection: Rejection) -> Result<Response, R
         TxTooBig => return unhandled(rejection),
         Config(_err) => return unhandled(rejection),
         PushIntoQueue(_err) => return unhandled(rejection),
-        Status(_err) => return unhandled(rejection),
         ConfigurationReload(_err) => return unhandled(rejection),
+        #[cfg(feature = "telemetry")]
+        Status(_err) => return unhandled(rejection),
+        #[cfg(feature = "telemetry")]
         Prometheus(_err) => return unhandled(rejection),
```
Comment (mversic) [inline cli/src/torii/routing.rs:340]: just a side note, how does the conditional compilation affect enum layout and subsequently enum serialization (e.g. for parity scale)? Is this something to consider? What do you think?

File: cli/src/lib.rs [hunk 29]
```diff
@@ -0,0 +1,319 @@
...
+        config: Configuration,
+        instruction_validator: IsInstructionAllowedBoxed<K::World>,
+        query_validator: IsQueryAllowedBoxed<K::World>,
+        broker: Broker,
+    ) -> Result<Self> {
+        if !config.disable_panic_terminal_colors {
+            color_eyre::install()?;
+        }
+        let telemetry = iroha_logger::init(&config.logger)?;
+        iroha_logger::info!("Hyperledgerいろは2にようこそ！");
```
Comment (mversic) [inline cli/src/lib.rs:147]: is everything ok with these whitespace characters. They look like squares on my browser

File: core/benches/validation.rs [hunk 30]
```diff
@@ -58,12 +61,12 @@ fn build_test_transaction(keys: KeyPair) -> Transaction {
 fn build_test_wsv(keys: KeyPair) -> WorldStateView<World> {
     WorldStateView::new({
         let mut domains = BTreeMap::new();
-        let mut domain = Domain::test(START_DOMAIN);
-        let account_id = AccountId::test(START_ACCOUNT, START_DOMAIN);
+        let mut domain = Domain::new(DomainId::new(START_DOMAIN).expect("is valid"));
```
Comment (Arjentix) [inline core/benches/validation.rs:64]: Everywhere outside this file you started using just `expect("Valid")`. May be you should also do it here for consistency?

File: permissions_validators/src/public_blockchain/transfer.rs [hunk 31]
```diff
@@ -2,9 +2,10 @@
 
 use super::*;
 
+#[allow(clippy::expect_used)]
 /// Can transfer user's assets permission token name.
 pub static CAN_TRANSFER_USER_ASSETS_TOKEN: Lazy<Name> =
-    Lazy::new(|| Name::test("can_transfer_user_assets"));
+    Lazy::new(|| Name::new("can_transfer_user_assets").expect("Tested. Works."));
```
[the remark this answers is not in the document]
Comment (mversic) [reply permissions_validators/src/public_blockchain/transfer.rs:8]: if that's the case, could we make a note of that somewhere?

File: data_model/tests/data_model.rs [hunk 32]
```diff
@@ -2,45 +2,57 @@
 
 use std::{str::FromStr, thread, time::Duration};
 
+use iroha::samples::*;
```
Comment (mversic) [inline data_model/tests/data_model.rs:5]: seems kinds weird to include samples?

File: cli/src/lib.rs (hunk 29 above)
Comment (s8sato) [reply cli/src/lib.rs:147]: image
Saying "Welcome to Hyperledger Iroha 2!".
I was wondering why only here is Japanese. Should be English

**Review by Arjentix — DISMISSED** [review]

File: client/Cargo.toml (hunk 27 above)
[the remark this answers is not in the document]
Comment (mversic) [reply client/Cargo.toml:35]: it used to be that we needed feature which was merged to `smallstr` master but wasn't included in the latest version. Now since the `0.3.0` was released we don't need to reference the specific commit anymore

File: cli/src/lib.rs (hunk 29 above)
Comment (mversic) [reply cli/src/lib.rs:147]: oh, I think that's fine then. I'd leave it in Japanese, I like that Iroha includes lots of references to Japanese culture.

File: data_model/src/peer.rs [hunk 33]
```diff
@@ -27,7 +27,7 @@ pub struct Peer {
 ///
 /// Equality is tested by `public_key` field only.
 /// Each peer should have a unique public key.
-#[derive(Debug, Clone, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+#[derive(Debug, Default, Clone, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply data_model/src/peer.rs:30]: Consequences of addressing one of @s8sato 's comments. A default is well defined, since the defaults of all constituent fields are also defined.

File: data_model/Cargo.toml [hunk 34]
```diff
@@ -21,6 +21,7 @@ default = ["std"]
 std = ["iroha_macro/std", "iroha_version/std", "iroha_version/warp", "iroha_crypto/std", "iroha_data_primitives/std", "thiserror"]
 # https://github.com/rust-lang/cargo/issues/6915
 roles = []
+cross_crate_testing = []
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply data_model/Cargo.toml:24]: I would still keep it around, just like `dex`.

File: permissions_validators/src/public_blockchain/transfer.rs (hunk 31 above)
[the remark this answers is not in the document]
Comment (appetrosyan) [reply permissions_validators/src/public_blockchain/transfer.rs:8]: I think you also need to be able to call certain other destructors before that's possible. I'd appreciate if you created an issue.

File: client/src/lib.rs (hunk 8 above)
Comment (appetrosyan) [reply client/src/lib.rs:25]: I want to disable unwrap globally. 

We only have expect for cases when an invariant is enforced and so something either never panics, or if it does, points to the violated invariant. Unwrap gives you no such information, and I think we can live without it. 

In tests the panic message should point you to the invariant, even if the invariant is "this string is a valid name", hence `.expect("valid")`.

File: cli/src/torii/routing.rs (hunk 28 above)
Comment (appetrosyan) [reply cli/src/torii/routing.rs:340]: Created #1976. Should discuss there

File: cli/src/lib.rs (hunk 29 above)
Comment (appetrosyan) [reply cli/src/lib.rs:147]: It's a minimal flourish that adds lots of flavour. I'll add the translation, but keep the japanese version.

File: data_model/tests/data_model.rs (hunk 32 above)
Comment (appetrosyan) [reply data_model/tests/data_model.rs:5]: Not particularly. It contains sample configuration files, genesis blocks and many other things that are provided *as* samples, so having them included in the tests is perfectly fine.

File: client/Cargo.toml (hunk 27 above)
[the remark this answers is not in the document]
Comment (appetrosyan) [reply client/Cargo.toml:35]: It was released 5 days ago. I'll update right now.

File: client/tests/integration/multiple_blocks_created.rs [hunk 35]
```diff
@@ -28,7 +32,7 @@ fn long_multiple_blocks_created() {
         )
         .into(),
     ));
-    let asset_definition_id = AssetDefinitionId::test("xor", "domain");
+    let asset_definition_id = AssetDefinitionId::new("xor", "domain").expect("Valid");
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply client/tests/integration/multiple_blocks_created.rs:35]: Yes. 
1) I want to get rid of `unwrap` entirely. `expect` will still cause an error, but one that you can `#[allow]`. 
2) expect actually gives you the invariant that must be violated for the thing to panic. In this case the `"Valid"` refers to the arguments. If this panics, you know what's wrong without looking at docs.

File: permissions_validators/src/public_blockchain/transfer.rs (hunk 31 above)
[the remark this answers is not in the document]
Comment (mversic) [reply permissions_validators/src/public_blockchain/transfer.rs:8]: created an issue

[the remark this answers is not in the document]
Comment (appetrosyan) [reply permissions_validators/src/public_blockchain/transfer.rs:8]: created #1978. Resolving.

File: core/src/stream.rs [hunk 36]
```diff
@@ -151,8 +151,8 @@ impl<M: DecodeVersioned> Stream<M> for warp::ws::WebSocket {
     type Message = warp::ws::Message;
 }
 
-#[cfg(test)]
-mod tests {
+#[cfg(feature = "cross_crate_testing")]
```
[the remark this answers is not in the document]
Comment (appetrosyan) [reply core/src/stream.rs:154]: For now yes.

**Review by mversic — DISMISSED** [review]

**Review by mversic — APPROVED** [review]

File: core/benches/validation.rs [hunk 37]
```diff
@@ -58,12 +61,12 @@ fn build_test_transaction(keys: KeyPair) -> Transaction {
...
         let mut domains = BTreeMap::new();
-        let mut domain = Domain::test(START_DOMAIN);
-        let account_id = AccountId::test(START_ACCOUNT, START_DOMAIN);
+        let mut domain = Domain::new(DomainId::new(START_DOMAIN).expect("Valid"));
+        let account_id = AccountId::new(START_ACCOUNT, START_DOMAIN).expect("Valid");
         let mut account = Account::new(account_id.clone());
         account.signatories.push(keys.public_key);
         domain.accounts.insert(account_id, account);
-        domains.insert(DomainId::test(START_DOMAIN), domain);
+        domains.insert(DomainId::new(START_DOMAIN).expect("is valid"), domain);
```
Comment (Arjentix) [inline core/benches/validation.rs:69]:
```suggestion
        domains.insert(DomainId::new(START_DOMAIN).expect("Valid"), domain);
```

**Review by Arjentix — APPROVED** [review]
